libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java

```diff
@@ -135,6 +149,47 @@ public static String getSHA256Hash(String privateKey) {
         return hashedString;
     }
 
+    /**
+     * Generate a keypair and get the encoded public key string.
+     *
+     * @param name Alias of the entry in which the generated key will appear in Android KeyStore.
+     * @return RSA public key string.
+     */
+    public static synchronized String getRSAPublicString(String name, int length) {
+        final SharedPreferences prefs = SalesforceSDKManager.getInstance().getAppContext().getSharedPreferences(ASYMM_SHARED_PREF_FILE, 0);
+        final String id = prefs.getString(getSharedPrefKey(name), null);
+
+        if (id != null) {
+            RSA_IDS.put(name, id);
+            return id;
+        } else {
+            String publicKeyBase64 = null;
+            KeyPair keyPair = getKeyPair(name, length);
+            if (keyPair != null) {
+                PublicKey publicKey = keyPair.getPublic();
+                publicKeyBase64 = Base64.encodeToString(publicKey.getEncoded(), Base64.NO_WRAP);
```

Comment:
Use `Base64.NO_WRAP | Base64.NO_PADDING | Base64.URL_SAFE` in the flags here, since we will be sending this to the server.